agent/pipeline_uploader_test.go

```diff
@@ -140,9 +140,54 @@ func TestFallbackPipelineUpload(t *testing.T) {
 	ctx := context.Background()
 	l := clicommand.CreateLogger(&clicommand.PipelineUploadConfig{LogLevel: "notice"})
 
-	jobID := api.NewUUID()
-	stepUploadUUID := api.NewUUID()
-	pipelineStr := `---
+	genSleeps := func(n int, s time.Duration) []time.Duration {
+		sleeps := make([]time.Duration, 0, n)
+		for i := 0; i < n; i++ {
+			sleeps = append(sleeps, s)
+		}
+		return sleeps
+	}
+
+	for _, test := range []struct {
+		name            string
+		num529s         int
+		expectedSleeps  []time.Duration
+		expectedUploads int
+		errStatus       *int
+	}{
+		{
+			name:            "happy",
+			num529s:         0,
+			expectedSleeps:  []time.Duration{},
+			expectedUploads: 1,
+		},
+		{
+			name:            "59_529s",
+			num529s:         59,
+			expectedSleeps:  genSleeps(58, 5*time.Second),
+			expectedUploads: 59,
+		},
+		{
+			name:            "60_529s",
+			num529s:         60,
+			expectedSleeps:  genSleeps(59, 5*time.Second),
+			expectedUploads: 60,
+		},
+		{
+			name:            "61_529s",
+			num529s:         61,
+			expectedSleeps:  genSleeps(59, 5*time.Second),
+			expectedUploads: 60,
+			errStatus:       func(i int) *int { return &i }(529),
```

Comment:
ahaha. 😬

What if `errStatus` was typed `int`, and only compared if it is != 0 ?

Reply:
I was debating whether to do this instead, but then I did not :)